app/recipe/tests/test_recipe_api.py

```diff
@@ -0,0 +1,207 @@
+            payload['name'],
+            Recipe.objects.get(id=response.data['id']).name
+        )
+        self.assertEquals(
+            len(response.data['ingredients']),
+            len(payload['ingredients'])
+        )
+
+    def test_get_recipes(self):
+        """Test retrieving a recipe"""
+        create_sample_recipe(
+            name='Roast Dinner',
+            description='Roasted potatoes and chicken'
+                        ' with vegetables and gravy.'
+        )
+
+        create_sample_recipe(
+            name='Beans on Toast',
+            description='Just the best.'
+        )
+
+        response = self.client.get(RECIPE_URL)
+
+        recipes = Recipe.objects.all().order_by('-name')
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), len(recipes))
+
+    def test_get_recipe(self):
+        """Test retrieving a single recipe using name as filter"""
+        test_recipe_name = 'Beans on Toast'
+
+        create_sample_recipe(
+            name='Roast Dinner',
+            description='Roasted potatoes and chicken'
+                        ' with vegetables and gravy.'
+        )
+
+        create_sample_recipe(
+            name=test_recipe_name,
+            description='Just the best recipe.'
+        )
+
+        response = self.client.get(RECIPE_URL, {'name': test_recipe_name})
+
+        recipes = Recipe.objects.all().order_by('-name')
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertNotEqual(len(response.data), len(recipes))
+        self.assertEqual(response.data[0]['name'], test_recipe_name)
+
+    def test_update_recipe(self):
+        """Test updating a recipe"""
+        self.recipe = create_sample_recipe(
+            name='Roast Dinner',
+            description='Roasted potatoes and chicken'
+                        ' with vegetables and gravy.'
+        )
+
+        payload = {
+            'name': 'Vegan Roast Dinner',
+            'description': 'Roasted potatoes and mushroom wellington'
+                           ' with vegetables and gravy.'
+        }
+
+        response = self.client.patch(
+            recipe_url(self.recipe.id),
+            payload, format='json'
+        )
+
+        self.recipe.refresh_from_db()
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(self.recipe.name, response.data['name'])
+        self.assertEqual(self.recipe.description, response.data['description'])
+
+    def test_delete_recipe(self):
+        """Test deleting a recipe"""
+        self.recipe = create_sample_recipe(
+            name='Carrot Cake',
+            description='Sponge cake with hella carrots.'
+        )
+
+        response = self.client.delete(
+            recipe_url(self.recipe.id),
+            format='json'
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertFalse(Recipe.objects.all())
```

Comment:
```suggestion
        self.assertFalse(Recipe.objects.exists())
```
Faster in case 1M recipes do still exist :sweat_smile: